### PR TITLE
Weapons Pack.

### PR DESCRIPTION
**Pull request by JohnnyBazooka89**

### CODE REVIEW

File: src/main/java/thePackmaster/util/CombatUtils.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package thePackmaster.util;
+
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.rooms.AbstractRoom;
+
+public class CombatUtils {
```
Comment (erasels) [inline src/main/java/thePackmaster/util/CombatUtils.java:7]: Delete this

File: src/main/java/thePackmaster/powers/weaponspack/WeaponPower.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+public class WeaponPower extends AbstractPackmasterPower {
+
+    public static final String POWER_ID = makeID("WeaponPower");
+    private static final PowerStrings powerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID);
+    public static final String NAME = powerStrings.NAME;
+    public static final String[] DESCRIPTIONS = powerStrings.DESCRIPTIONS;
+
+    public WeaponPower(AbstractCreature owner) {
+        super(POWER_ID, NAME, PowerType.BUFF, false, owner, -1);
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/weaponspack/WeaponPower.java:30]: This is just a system power, it'd be nice if you used the neutral type so it doesn't interact with other powers. Could also make this invisible by implementing the InvisiblePower interface from StSLib
```suggestion
        super(POWER_ID, NAME, NeutralPowertypePatch.NEUTRAL, false, owner, -1);
```

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/cards/weaponspack/AgileStrike.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    private static final int UPGRADE_PLUS_ATTACK_DMG = 4;
+
+    public AgileStrike() {
+        super(ID, COST, CardType.ATTACK, CardRarity.UNCOMMON, CardTarget.ENEMY);
+
+        this.baseDamage = this.damage = ATTACK_DMG;
+
+        this.tags.add(CardTags.STRIKE);
+
+        if (CombatUtils.isIndeedWithoutADoubtInCombat()) {
```
[the remark this answers is not in the document]
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/cards/weaponspack/AgileStrike.java]: Changed.

File: src/main/java/thePackmaster/cards/weaponspack/AgileStrike.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        super.applyPowers();
+    }
+
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        AbstractDungeon.actionManager.addToBottom(new DamageAction(m, new DamageInfo(p, this.damage, this.damageTypeForTurn), AbstractGameAction.AttackEffect.SLASH_DIAGONAL));
+    }
+
+    private void reduceCostToZeroForATurnIfNeeded() {
+        boolean shouldCostBeReduced = false;
```
[the remark this answers is not in the document]
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/cards/weaponspack/AgileStrike.java:53]: Added that line.

File: src/main/java/thePackmaster/cards/weaponspack/Sword.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    public void upgrade() {
+        if (!this.upgraded) {
+            this.upgradeName();
+            this.upgradeMagicNumber(UPGRADE_PLUS_WEAPON_DURABILITY);
+        }
+    }
+
+    @Override
+    public void initializeDescription() {
+        super.initializeDescription();
```
[the remark this answers is not in the document]
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/cards/weaponspack/Sword.java]: Hmm. Looks not needed, it's the same in the blackbeard mod :P and it was copied over.

Removed.

File: src/main/java/thePackmaster/cards/weaponspack/SwordOfChaos.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import thePackmaster.actions.weaponspack.EquipAction;
+import thePackmaster.orbs.weaponspack.SwordOfChaosOrb;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+public class SwordOfChaos extends AbstractWeaponsPackCard {
+
+    public static final String ID = makeID("SwordOfChaos");
+    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
```
[the remark this answers is not in the document]
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/cards/weaponspack/SwordOfChaos.java]: Yeah, deleted.

File: src/main/java/thePackmaster/patches/weaponspack/SwordDiscoveryPatch.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.util.ArrayList;
+
+@SpirePatch(clz = CardRewardScreen.class, method = "discoveryOpen", paramtypez = {})
+public class SwordDiscoveryPatch {
+    public static boolean swordDiscovery;
+    public static boolean upgraded;
+
+    @SpireInsertPatch(locator = Locator.class, localvars = {"derp"})
+    public static void Insert(CardRewardScreen cardRewardScreen, ArrayList<AbstractCard> derp) {
```
[the remark this answers is not in the document]
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/patches/weaponspack/SwordDiscoveryPatch.java]: I refactored this to work like Wish. The Blackbeard was created before the Watcher, so the code was a little bit outdated for this card.

File: src/main/java/thePackmaster/util/CombatUtils.java (hunk 1 above)
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/util/CombatUtils.java]: Deleted.

File: src/main/java/thePackmaster/powers/weaponspack/WeaponPower.java (hunk 2 above)
Comment (JohnnyBazooka89) [reply src/main/java/thePackmaster/powers/weaponspack/WeaponPower.java]: Changed to Neutral and made it invisible.

**Pull request merged**